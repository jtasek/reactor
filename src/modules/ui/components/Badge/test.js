import React from 'react'
import Badge from './index'
import renderer from 'react-test-renderer'

describe('<Badge />', () => {
  it('should render without crashing', () => {
    const component = renderer.create(
      <Badge />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
