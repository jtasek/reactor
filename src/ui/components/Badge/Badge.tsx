import React, { FC } from 'react';

export interface Props {
  text: string;
  color: string;
  size: number;
}

export const Badge: FC<Props> = ({ text, color, size }) => (
  <span
    style={{
      backgroundColor: color,
      borderRadius: `${size}px`,
      color: 'white',
      display: 'inline-block',
      fontSize: 'large',
      fontWeight: 700,
      height: `${size}px`,
      left: 0,
      minHeight: `${size}px`,
      minWidth: `${size}px`,
      position: 'relative',
      textAlign: 'center',
      top: 0,
      verticalAlign: 'middle',
      whiteSpace: 'nowrap',
      width: `${size}px`
    }}
  >
    {text}
  </span>
);
