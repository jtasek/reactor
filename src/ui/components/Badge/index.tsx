import React, { FC } from 'react';

import { Size } from '../../../app/types';

interface Props {
  text: string;
  color: string;
  size: Size;
}

export const Badge: FC<Props> = ({ text, color, size }) => (
  <span
    style={{
      borderRadius: '50%',
      color: color,
      height: size.height,
      position: 'absolute',
      right: 0,
      top: 0,
      width: size.width
    }}
  >
    {text}
  </span>
);
