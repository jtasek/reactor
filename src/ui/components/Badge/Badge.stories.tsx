import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import { Badge, Props } from './Badge';

export default {
  title: 'Example/Badge',
  component: Badge
} as Meta;

const Template: Story<Props> = (args) => <Badge {...args} />;

export const Small = Template.bind({});
Small.args = {
  text: '1',
  color: 'red',
  size: 15
};

export const Medium = Template.bind({});
Medium.args = {
  text: '1',
  color: 'red',
  size: 20
};

export const Large = Template.bind({});
Large.args = {
  text: '1',
  color: 'red',
  size: 25
};
