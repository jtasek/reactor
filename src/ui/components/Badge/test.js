import React from 'react';
import Badge from './index';
import renderer from 'react-test-renderer';

describe('<Badge />', () => {
  it('should render without crashing', () => {
    const component = renderer.create(<Badge />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
