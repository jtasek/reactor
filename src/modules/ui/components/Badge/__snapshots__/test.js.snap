// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render without crashing 1`] = `
<span
  style={
    Object {
      "borderRadius": "50%",
      "color": undefined,
      "height": undefined,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "width": undefined,
    }
  }
/>
`;
